libs/medic-users-meta/migrations/202304140054.do.telemetry_metrics.sql

```diff
@@ -0,0 +1,47 @@
+DROP MATERIALIZED VIEW IF EXISTS useview_telemetry_metrics;
+
+CREATE MATERIALIZED VIEW public.useview_telemetry_metrics
+TABLESPACE pg_default AS 
+
+WITH telemetry_docs_with_metric_blob AS (
+ SELECT
+    doc #>> '{metadata,deviceId}' AS device_id,
+    doc #>> '{_id}' AS telemetry_doc_id,
+    doc #>> '{metadata,user}' AS user_name,
+    concat_ws(
+      '-', doc #>> '{metadata,year}',
+      CASE
+          WHEN (doc #>> '{metadata,day}') IS NULL AND ((doc #>> '{metadata,versions,app}') IS NULL OR string_to_array("substring"(doc #>> '{metadata,versions,app}', '(\d+.\d+.\d+)'::text), '.'::text)::integer[] < '{3,8,0}'::integer[]) THEN ((doc #>> '{metadata,month}')::integer) + 1
```

Comment:
For some reason I thought I've seen poorly named branches that don't have version they were based off of in the name. If that's not possible, ignore this comment. If that sounds possible, read-on.

If `'{metadata,versions,app}'` is a poorly named branch, I think the `ELSE` will catch it and just use `'{metadata,month}'` without incrementing, which will be correct sometimes and incorrect sometimes, depending on if that branch was before or after 3.8.0. At this point we really have no clue though. One possible solution would be to just make sure we never have a 0 month, even though it might not be right. The other option is to let the calling query fail so that the person trying to use this matview knows there are some inconsistencies and deals with it in their calling query. Bad branch names are maybe rare enough (or non-existent) that it's not worth dealing with? In those scenarios, you could have non-zero months that are not right but won't fail anyway. ¯\_(ツ)_/¯

Reply:
this is just so messy. i'm not sure what to do.

Reply:
Actually, I just realized the `period_start` field is being cast to a date in the matview anyway so it'll just blow up here if it comes across a month = 0, not later down the road.

Anyway, yeah. `purge-specfic-tasks-3.9.x` is a branch in one project, though I think in that case it should actually be fine because it should have the correct month index since it is > 3.8. 

Either way, it's really rare, anything pre 3.8 is going to be really old data and probably not useful for analysis anyway, so personally I think it's probably just best to make sure it doesn't crash.